sql/tablemetadatacache: TestUpdateTableMetadataCacheAutomaticUpdates failed

sql/tablemetadatacache.TestUpdateTableMetadataCacheAutomaticUpdates [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17784002?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17784002?buildTab=artifacts#/) on release-24.3 @ [7a032e5d942f0ba8febfffaf233dde92af6b4ea8](https://github.com/cockroachdb/cockroach/commits/7a032e5d942f0ba8febfffaf233dde92af6b4ea8):


```
=== RUN   TestUpdateTableMetadataCacheAutomaticUpdates
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/123ee902c881bcac495467ae4eaf7f18/logTestUpdateTableMetadataCacheAutomaticUpdates2587333503
    test_log_scope.go:76: use -show-logs to present logs inline
    update_table_metadata_cache_job_test.go:118: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/123ee902c881bcac495467ae4eaf7f18/logTestUpdateTableMetadataCacheAutomaticUpdates2587333503
--- FAIL: TestUpdateTableMetadataCacheAutomaticUpdates (64.98s)
=== RUN   TestUpdateTableMetadataCacheAutomaticUpdates/AutomaticUpdatesEnabled
    update_table_metadata_cache_job_test.go:95: 
        	Error Trace:	pkg/sql/tablemetadatacache/update_table_metadata_cache_job_test.go:95
        	Error:      	Received unexpected error:
        	            	timed out waiting for job run 1. auto_updates_enabled:true, data_valid_duration: 00:00:00.05
        	Test:       	TestUpdateTableMetadataCacheAutomaticUpdates/AutomaticUpdatesEnabled
        	Messages:   	Job should have run at least 3 times
    --- FAIL: TestUpdateTableMetadataCacheAutomaticUpdates/AutomaticUpdatesEnabled (30.01s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestUpdateTableMetadataCacheAutomaticUpdates.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-44453

--- a/pkg/sql/tablemetadatacache/update_table_metadata_cache_job_test.go
+++ b/pkg/sql/tablemetadatacache/update_table_metadata_cache_job_test.go
@@ -85,12 +85,13 @@ func TestUpdateTableMetadataCacheAutomaticUpdates(t *testing.T) {
 	DataValidDurationSetting.SetOnChange(&s.ClusterSettings().SV, func(ctx context.Context) {
 		log.Infof(ctx, "Updating data valid duration setting to %s",
 			DataValidDurationSetting.Get(&s.ClusterSettings().SV))
+
 	})
+	DataValidDurationSetting.Override(ctx, &s.ClusterSettings().SV, 50*time.Millisecond)
 	require.Zero(t, len(updater.mockCalls), "Job should not run automatically by default")
 
 	t.Run("AutomaticUpdatesEnabled", func(t *testing.T) {
 		conn.Exec(t, `SET CLUSTER SETTING obs.tablemetadata.automatic_updates.enabled = true`)
-		DataValidDurationSetting.Override(ctx, &s.ClusterSettings().SV, 50*time.Millisecond)
 		err := waitForJobRuns(3, 30*time.Second)
 		require.NoError(t, err, "Job should have run at least 3 times")
 		require.GreaterOrEqual(t, len(updater.mockCalls), 3, "Job should have run at least 3 times")
@@ -109,7 +110,6 @@ func TestUpdateTableMetadataCacheAutomaticUpdates(t *testing.T) {
 
 	t.Run("AutomaticUpdatesDisabled", func(t *testing.T) {
 		conn.Exec(t, `SET CLUSTER SETTING obs.tablemetadata.automatic_updates.enabled = f`)
-		DataValidDurationSetting.Override(ctx, &s.ClusterSettings().SV, 50*time.Millisecond)
 		initialCount := len(updater.mockCalls)
 		err := waitForJobRuns(1, 200*time.Millisecond)
 		require.Error(t, err, "Job should not run after being disabled")

